src/PinAssignment/PortsLoader.cpp

```diff
@@ -20,8 +20,52 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 #include "PortsLoader.h"
 
+#include <QFile>
+#include <iostream>
+
+#include "nlohmann_json/json.hpp"
+using json = nlohmann::ordered_json;
+
 namespace FOEDAG {
 
-PortsLoader::PortsLoader() {}
+PortsLoader::PortsLoader(PortsModel *model, QObject *parent)
+    : QObject(parent), m_model(model) {}
+
+PortsLoader::~PortsLoader() {}
+
+bool PortsLoader::load(const QString &file) {
+  if (!m_model) return false;
+  QFile f{file};
+  if (!f.open(QFile::ReadOnly)) return false;
+
+  QString content = f.readAll();
+  json jsonObject;
+  try {
+    jsonObject = json::parse(content.toStdString());
+  } catch (json::parse_error &e) {
+    // output exception information
+    std::cerr << "Json Error: " << e.what() << '\n'
```

Comment:
Use logger here

Reply:
It seems like we don't have infrastructure for logging outside the compiler. At least, I don't have a quick fix

Reply:
Sounds good, let's connect offline on this.